src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -723,13 +723,10 @@ private Task WriteSuffix()
             var end = SocketOutput.ProducingStart();
             if (_keepAlive && hasConnection)
             {
-                foreach (var connectionValue in responseHeaders.HeaderConnection)
+                var connectionValue = responseHeaders.HeaderConnection.ToString();
```

Comment:
Following the [pattern we use to check the request Connection header](https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/MessageBody.cs#L107) for whether or not it's been set to "keep-alive". Super cheap in the 99.9...% case where either it hasn't been set (and `hasConnection` is false) or the `Connection` header has a single value.